### DISCUSSION TITLE
files (debug.cfg, debug.custom.json, esp.svd) littering library example folders

### DISCUSSION THREAD

**Discussion by Makuna** [espressif/arduino-esp32#7784] [Q&A]
I am running into a little issue with the latest release with the Arduino IDE.
Whenever I load an example from a library and build, the example directory in the library gets littered with files (debug.cfg, debug.custom.json, esp.svd). The examples directory for a library is supposed to be read-only.
This is primarily a problem because as a developer for libraries, my github is mapped to the library location so it shows extra files all the time that I need to "discard" before submitting changes.

Should these files only be saved into the working build directory like the obj?
Should these files only be saved when you actually select to save the example sketch file?

**Comment by me-no-dev** [comment]
This is already fixed in master and will be part of the upcoming 2.0.7. Files will be copied only if debug adapter is selected in the board menu
